src/Microsoft.DotNet.Interactive/KernelInvocationContext.cs

```diff
@@ -47,16 +51,22 @@ private KernelInvocationContext(KernelCommand command)
             category: nameof(KernelInvocationContext),
             logOnStart: true);
 
-        _cancellationTokenSource = new CancellationTokenSource();
+        _cancellationTokenSource =
+            _cancellationTokenSources.GetOrAdd(
+                command.GetOrCreateToken(),
+                s =>
+                {
+                    _ownsCancellationTokenSource = true;
```

Comment:
Does this assume the first command we see here is the original/parent?

Reply:
the assumption is : **the first one to be used to establish will be the parent**